lib/system.js

```diff
@@ -68,7 +70,13 @@ module.exports = class SystemView extends ReadyResource {
   }
 
   async checkout (length) {
-    const checkout = new SystemView(this.core.session(), length)
+    const checkout = new SystemView(
+      this.core.session(),
+      {
+        checkout: length,
+        maxCacheSize: this.maxCacheSize
```

Comment:
dont think this prop is on the system, you should get it from the existing db

Reply:
You're right, well-spotted, I fixed it.

I think that method is never used though (from what I can see, it's never used in autobase itself, nor is it tested, and I don't think it makes sense to access it directly from outside autobase), so might make sense to just remove it in a follow-up PR, or perhaps add a test

Reply:
ahh I think checkout will stick around for a while, it's quite useful to have